api/plugin.go

```diff
@@ -240,3 +239,36 @@ func calculateTransactionHash(txData string) (string, error) {
 	hash := tx.Hash().Hex()[2:]
 	return hash, nil
 }
+
+// right now, this functon if for payroll plugin on evm chains
+// TODO: add support for other plugins and be chain agnostic
+func generateTransactionHash(payload *v1.KeysignPayload) (string, error) {
```

Comment:
ok for testing, but this will eventually go into the plugin interface - will send across the diagram on discord shortly